packages/react-layouts/src/Modal/index.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright (c) Quid, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+// @flow
+
+import * as React from 'react';
+import ReactModal from 'react-modal';
+import styled from '@emotion/styled/macro';
+import Centerer from './Centerer';
+import { withFallback as wf } from '@quid/theme';
+import Color from 'color';
+import { ClassNames } from '@emotion/core';
+import Header from './Header';
+import { ACTION } from './importanceTypes';
+import type { Importance } from './importanceTypes';
+import ActionBar from './ActionBar';
+import css from '@emotion/css/macro';
+
+type Props = {
+  title?: React.Node,
+  alt?: string, // specify if `title` is not a string
+  icon?: string,
+  noPadding?: boolean,
+  importance?: Importance,
+  ignoreHeightLimits?: boolean,
+  centerVertically?: boolean,
+  action?: React.Node,
+  renderActionLeft?: string => React.Node,
+  renderActionRight?: string => React.Node,
+  children: React.Node,
+  overlayClassName?: string,
+  styles?: Object,
+  ariaHideApp?: boolean,
+  className?: string,
+  isForm?: boolean,
+  formProps?: { [string]: * },
+};
+
+const OptionalForm = styled(({ isForm, children, ...props }) => {
+  const Tag = isForm ? 'form' : 'div';
+  return <Tag {...props}>{children}</Tag>;
+})`
+  flex: 1;
+  display: flex;
+  flex-direction: column;
+  overflow-y: auto;
+`;
+
+const Main = styled.main`
+  min-height: inherit;
+  flex: 1;
+`;
+
+const reduceOpacity = (color, alpha) =>
+  Color(color)
+    .alpha(alpha)
+    .string();
+
+const Modal = styled(
+  ({
+    isForm,
+    title,
+    alt,
+    icon,
+    noPadding = false,
+    importance = ACTION,
+    ignoreHeightLimits = false,
+    centerVertically = false,
+    children,
+    action,
+    renderActionLeft,
+    renderActionRight,
+    overlayClassName,
+    ariaHideApp = false,
+    formProps,
+    ...props
+  }: Props) => {
+    if (title && typeof title !== 'string' && !alt) {
+      throw new Error(
+        "`alt` property is required when `title` isn't a string!"
+      );
+    }
+
+    return (
+      <ClassNames>
+        {({ css, theme }) => (
+          <ReactModal
+            overlayClassName={css`
```

Comment:
Do we want this to be customizable from the outside?

If so, we need to do `cx(css`...`, overlayClassName)`. If not, we should remove `overlayClassName` from the props